Add StepDownPrimary to the replicaset methods.

jameinel opened the pull request:
This will ask the Primary to step down using the default parameters (60s
the primary should not become primary again, and 10s for a secondary to
become up-to-date, and not forcing the step down if a secondary fails to
get up to date.)

We also suppress the io.EOF because we expect all sessions to get
disconnected. Anything calling this function needs to be prepared for
that to happen, but it is expected, not an error.

jameinel committed:
Add StepDownPrimary to the replicaset methods.

This will ask the Primary to step down using the default parameters (60s
the primary should not become primary again, and 10s for a secondary to
become up-to-date, and not forcing the step down if a secondary fails to
get up to date.)

We also suppress the io.EOF because we expect all sessions to get
disconnected. Anything calling this function needs to be prepared for
that to happen, but it is expected, not an error.

jameinel commented:
![](https://github.trello.services/images/mini-trello-icon.png) [replicaset StepDownPrimary](https://trello.com/c/xXdI7Y1N/168-replicaset-stepdownprimary)

jameinel committed:
Include dependencies.tsv so we can run a proper build bot.

manadart commented on replicaset.go:
```diff
@@ -421,6 +421,27 @@ type Config struct {
+       // replSetStepDown can take a number of arguments, such as "number of seconds to wait for a secondary to become
+       // up-to-date", "how long to force the current primary to not be primary", and "force" whether or not the primary
+       // should be demoted right now. But we just use the defaults for all of those.
+	   err := session.Run("replSetStepDown", nil)
```
Rest of the method has an odd 3-space indent...

jameinel replied:
thanks, just doing this on a different machine and forgot to run go fmt.

jameinel committed:
go fmt

jameinel commented:
$$merge$$

jameinel committed:
TearDownTest would panic if mongo wasn't found.

Instead, only Destroy the root if we succeed in setting it up.

jameinel commented:
$$merge$$ the bot was configured without mongo being installed.

jameinel committed:
Testing with multiple versions of Mongo shows some changes needed.

3.2 complains if you don't pass step down seconds. (It seems it treats
step down seconds as 0, and then complains because step down seconds is
less than the default 10s secondary catch up seconds.)

Also, Mongo2.4 returns 'nil' for tags, but Mongo 3.2 returns {} for
tags. So we have to set real tags so that both return the same values.

jameinel commented:
$$merge$$ I pushed up a small change. It seems 2.4 and 3.2 have slightly different behaviors (3.4 and 3.6 seem to follow 3.2 behavior.)
